### PR TITLE
LPS-165295  & LPS-165396 Fixed headings hierarchy in the site builder for Home page

### PR DESCRIPTION
**Pull request by veroglez**
Related Issue: https://issues.liferay.com/browse/LPS-165295
Related Issue: https://issues.liferay.com/browse/LPS-165396

cc @sandrodw3

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by veroglez** [comment]
ci:test:sf

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/UnsafeHTML.js [hunk 1]
```diff
@@ -141,6 +141,20 @@ export default class UnsafeHTML extends React.PureComponent {
 			ref.removeAttribute('id');
 		}
 
+		if (this.props.id) {
+			ref.id = this.props.id;
+		}
+		else {
+			ref.removeAttribute('id');
+		}
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/UnsafeHTML.js:149]: Ctrl+C, Ctrl+V, Ctrl+V

Comment (veroglez) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/UnsafeHTML.js]: 😆

**Comment by p2kmgcl** [comment]
ci:forward

**Comment by veroglez** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:forward

**Comment by ozysouza** [comment]
@veroglez, could you please fix the rebase error? I'll analyze the failures by the morning but I pretty sure we will have to manually send it, the unrelated failures on master are going to block the PR.

**Comment by ozysouza** [comment]
> Hi @ozysouza! done!

Thanks! Let's try to forward once more and if it doesn't work I will analyze the results. Thanks!

**Comment by ozysouza** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:stop

**Comment by ealonso** [comment]
ci:forward

**Comment by liulukiwi** [comment]
ci:forward

**Comment by veroglez** [comment]
ci:forward

**Comment by liulukiwi** [comment]
Hi @veroglez, could you cherry-pick my commits at https://github.com/liulukiwi/liferay-portal/commits/LPS-165396 for the test-fix? Besides, I find an issue when analysising the failures in former relevant results, Submit Button Missing warning occurs when publishing content page mapped to object, could you take a look? here are the steps:
1. Add a picklist
2. Add an object and add the created picklist to the field
3. Publish the object
4. Add a Form Container to a content page and then map it to the object
5. Publish the content page

Actual result:
Submit Button Missing warining occurs when publishing the page.

And [js-unit-jdk8](https://testray.liferay.com/reports/production/logs/2022-10/test-1-25/test-portal-acceptance-pullrequest(master)/5073/js-unit-jdk8/0/0/jenkins-console.txt.gz?authuser=0) also failed in this pull, could you take a look? thanks!
```
  [exec] Summary of all failing tests
     [exec] FAIL test/page_editor/app/components/UnsafeHTML.test.js
     [exec]   ??? UnsafeHTML ??? renders the given HTML markup
     [exec] 
     [exec]     TestingLibraryElementError: Unable to find an accessible element with the role "heading"
     [exec] 
     [exec]     There are no accessible roles. But there might be some inaccessible roles. If you wish to access them, then set the `hidden` option to `true`. Learn more about this here: https://testing-library.com/docs/dom-testing-library/api-queries#byrole
     [exec] 
     [exec]     Ignored nodes: comments, <script />, <style />
     [exec]     [36m<body>[39m
     [exec]       [36m<div>[39m
     [exec]         [36m<div[39m
     [exec]           [33maria-hidden[39m=[32m"true"[39m
     [exec]         [36m>[39m
     [exec]           [36m<h1>[39m
     [exec]             [0mHello [0m
     [exec]             [36m<strong>[39m
     [exec]               [0mG??rjen[0m
     [exec]             [36m</strong>[39m
     [exec]           [36m</h1>[39m
     [exec]         [36m</div>[39m
     [exec]       [36m</div>[39m
     [exec]     [36m</body>[39m
     [exec] 
     [exec]       27 | 		);
     [exec]       28 | 
     [exec]     > 29 | 		expect(getByRole('heading')).toBeInTheDocument();
     [exec]          | 		       ^
     [exec]       30 | 	});
     [exec]       31 | 
     [exec]       32 | 	it('allows adding any id', () => {
     [exec] 
     [exec]       at Object.getElementError (../../../node_modules/@testing-library/dom/dist/config.js:38:19)
     [exec]       at ../../../node_modules/@testing-library/dom/dist/query-helpers.js:90:38
     [exec]       at ../../../node_modules/@testing-library/dom/dist/query-helpers.js:62:17
     [exec]       at getByRole (../../../node_modules/@testing-library/dom/dist/query-helpers.js:111:19)
     [exec]       at Object.<anonymous> (test/page_editor/app/components/UnsafeHTML.test.js:29:10)
     [exec]       at processTicksAndRejections (node:internal/process/task_queues:96:5)
     [exec] 
     [exec]   ??? UnsafeHTML ??? allows using a custom HTML tag as container
     [exec] 
     [exec]     TestingLibraryElementError: Unable to find an accessible element with the role "heading"
     [exec] 
     [exec]     There are no accessible roles. But there might be some inaccessible roles. If you wish to access them, then set the `hidden` option to `true`. Learn more about this here: https://testing-library.com/docs/dom-testing-library/api-queries#byrole
     [exec] 
     [exec]     Ignored nodes: comments, <script />, <style />
     [exec]     [36m<body>[39m
     [exec]       [36m<div>[39m
     [exec]         [36m<h1[39m
     [exec]           [33maria-hidden[39m=[32m"true"[39m
     [exec]         [36m>[39m
     [exec]           [0mThe Title[0m
     [exec]         [36m</h1>[39m
     [exec]       [36m</div>[39m
     [exec]     [36m</body>[39m
     [exec] 
     [exec]       51 | 		);
     [exec]       52 | 
     [exec]     > 53 | 		expect(getByRole('heading')).toBeInTheDocument();
     [exec]          | 		       ^
     [exec]       54 | 	});
     [exec]       55 | 
     [exec]       56 | 	it('calls onRender prop whenever the content is updated', () => {
     [exec] 
     [exec]       at Object.getElementError (../../../node_modules/@testing-library/dom/dist/config.js:38:19)
     [exec]       at ../../../node_modules/@testing-library/dom/dist/query-helpers.js:90:38
     [exec]       at ../../../node_modules/@testing-library/dom/dist/query-helpers.js:62:17
     [exec]       at getByRole (../../../node_modules/@testing-library/dom/dist/query-helpers.js:111:19)
     [exec]       at Object.<anonymous> (test/page_editor/app/components/UnsafeHTML.test.js:53:10)
     [exec]       at processTicksAndRejections (node:internal/process/task_queues:96:5)
     [exec] 
     [exec] 
     [exec] Test Suites: 1 failed, 84 passed, 85 total
     [exec] Tests:       2 failed, 515 passed, 517 total
```

**Comment by veroglez** [comment]
Hi @liulukiwi! I have fixed the frontend tests. I will look at the issue you mention as soon as possible. Thanks!

**Comment by veroglez** [comment]
Hi @liulukiwi! I have updated the PR uploading the fix for the problem you mentioned, could you check that everything is ok? Thanks!

**Comment by veroglez** [comment]
ci:test:relevant

**Comment by ealonso** [comment]
ci:stop

**Comment by ealonso** [comment]
ci:forward

**Comment by liulukiwi** [comment]
ci:forward

**Comment by ealonso** [comment]
Just started reviewing :)

**Comment by ealonso** [comment]
ci:stop

**Comment by ealonso** [comment]
https://github.com/liferay-echo/liferay-portal/pull/10043

**Pull request closed without merging**